pkg/operator/ceph/object/objectstore.go

```diff
@@ -399,47 +399,47 @@ func createMultisite(objContext *Context, endpointArg string) error {
 	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
 
 	// create the realm if it doesn't exist yet
-	output, err := RunAdminCommandNoMultisite(objContext, true, "realm", "get", realmArg)
-	if err != nil {
+	output, getRealmErr := RunAdminCommandNoMultisite(objContext, true, "realm", "get", realmArg)
+	if getRealmErr != nil {
 		// ENOENT means "No such file or directory"
-		if code, err := exec.ExtractExitCode(err); err == nil && code == int(syscall.ENOENT) {
-			output, err = RunAdminCommandNoMultisite(objContext, false, "realm", "create", realmArg)
-			if err != nil {
-				return errorOrIsNotFound(err, "failed to create ceph realm %q, for reason %q", objContext.ZoneGroup, output)
+		if code, extractErr := exec.ExtractExitCode(getRealmErr); extractErr == nil && code == int(syscall.ENOENT) {
+			output, createRealmErr := RunAdminCommandNoMultisite(objContext, false, "realm", "create", realmArg)
+			if createRealmErr != nil {
+				return errorOrIsNotFound(createRealmErr, "failed to create ceph realm %q, for reason %q", objContext.ZoneGroup, output)
 			}
 			logger.Debugf("created realm %q", objContext.Realm)
```

Comment:
I think what Travis pointed out is right. I'm pretty sure this code should be modified to explicitly handle the case where `extractErr` is non-nil.